### PR TITLE
Add rest api

### PR DESCRIPTION
**Pull request by linjunjj**
### Description
Add rest api for support `clientid CURD `and add rest api test case, Introducing `emqx_ct_helper` in the test cases.

### CODE REVIEW

**Comment by tigercl** [comment]
Please update ReadMe

File: Makefile [hunk 1]
```diff
@@ -4,13 +4,18 @@ PROJECT_DESCRIPTION = EMQ X Authentication with ClientId/Password
...
 BRANCH := $(if $(filter $(CUR_BRANCH), master develop), $(CUR_BRANCH), develop)
 
-DEPS = emqx_passwd clique
+DEPS = emqx_passwd clique minirest
 dep_emqx_passwd = git-emqx https://github.com/emqx/emqx-passwd v1.0
 dep_clique      = git-emqx https://github.com/emqx/clique v0.3.11
+dep_minirest = git-emqx https://github.com/emqx/minirest v0.2.0
 
-BUILD_DEPS = emqx cuttlefish
+BUILD_DEPS = emqx cuttlefish emqx_management
```
Comment (tigercl) [inline Makefile]: Remove dep for emqx_management

File: src/emqx_auth_clientid_api.erl [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+return() ->
+    emqx_mgmt:return().
+
+return(R) ->
+    emqx_mgmt:return(R).
+
+format([{?APP, ClientId, Password}]) ->
+    [{clientid, ClientId},
+     {password, emqx_auth_clientid:unwap_salt(Password)}].
```
Comment (tigercl) [inline src/emqx_auth_clientid_api.erl]: unwap_salt -> unwrap_salt

File: src/emqx_auth_clientid.erl [hunk 3]
```diff
@@ -28,6 +28,7 @@
 
 -export([ init/1
         , check/2
+        , unwarp_salt/1
```
Comment (gilbertwong96) [inline src/emqx_auth_clientid.erl]: unwrap

File: src/emqx_auth_clientid.erl [hunk 4]
```diff
@@ -149,6 +150,9 @@ check(Credentials = #{client_id := ClientId, password := Password}, #{hash_type
             end
     end.
 
+unwarp_salt(<<_Salt:4/binary, HashPasswd/binary>>) ->
```
Comment (tigercl) [inline src/emqx_auth_clientid.erl]: unwarp_salt -> unwrap_salt

File: src/emqx_auth_clientid_api.erl [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+        ok -> return();
+        Error -> return(Error)
+    end.
+
+%%------------------------------------------------------------------------------
+%% Interval Funcs
+%%------------------------------------------------------------------------------
+
+return() ->
+    emqx_mgmt:return().
```
Comment (hjianbo) [inline src/emqx_auth_clientid_api.erl]: 1. Replace this fun with `minirest:return/x`
2. Update the minirest dependency to v0.2.2

File: Makefile [hunk 6]
```diff
@@ -4,14 +4,19 @@ PROJECT_DESCRIPTION = EMQ X Authentication with ClientId/Password
 CUR_BRANCH := $(shell git branch | grep -e "^*" | cut -d' ' -f 2)
 BRANCH := $(if $(filter $(CUR_BRANCH), master develop), $(CUR_BRANCH), develop)
 
-DEPS = emqx_passwd clique
+DEPS = emqx_passwd clique minirest
 dep_emqx_passwd = git-emqx https://github.com/emqx/emqx-passwd v1.0
 dep_clique      = git-emqx https://github.com/emqx/clique v0.3.11
+dep_minirest = git-emqx https://github.com/emqx/minirest v0.2.0
```
Comment (tigercl) [inline Makefile]: v0.2.0 -> v0.2.2

File: README.md [hunk 7]
```diff
@@ -25,6 +25,78 @@ etc/emqx_auth_clientid.conf:
 auth.client.password_hash = sha256
 ```
 
+[REST API](https://developer.emqx.io/docs/emq/v3/en/rest.html)
+------------
+
+List all clientids
+```
+#Request
```
Comment (tigercl) [inline README.md]: Add space between '#' and 'Request'

File: src/emqx_auth_clientid.erl [hunk 8]
```diff
@@ -28,6 +28,7 @@
 
 -export([ init/1
         , check/2
+        , unwrap_salt/1
```
Comment (tigercl) [inline src/emqx_auth_clientid.erl]: move unwrap_salt/1 to single export

**Review by hjianbo — APPROVED** [review]

**Pull request merged**